src/backend/distributed/commands/multi_copy.c

```diff
@@ -126,6 +126,134 @@
 #include "utils/tuplestore.h"
 #include "utils/memutils.h"
 
+/* Data structures from copy.c, to keep track of COPY processing state */
```

Comment:
We should work very hard avoiding copying all these. It'll be a maintenance nightmare to keep the structs in sync.

In fact, we can just check the target table for the problematic types / NextCopyFrom() returns values for all the columns in the target table anyway.